tests/ui/option_manual_map.fixed

```diff
@@ -0,0 +1,22 @@
+// run-rustfix
+
+#![warn(clippy::option_manual_map)]
+#![allow(clippy::map_identity)]
+
+fn main() {
+    Some(0).map(|_| 2);
+
+    Some(0).map(|x| x + 1);
+
+    Some("").map(|x| x.is_empty());
+
+    Some(0).map(|x| !x);
+
+    #[rustfmt::skip]
+    Some(0).map(std::convert::identity);
```

Comment:
Should we emit the lint in this line? It seems to emit the `map_identity` lint in this fixed code.

Reply:
The linted code was basically a long way of mapping identity. This is also not the only lint to suggest a change that triggers another lint.

This case is easy enough to catch though.